### PR TITLE
[vulkan] Reduce runtime host overhead

### PR DESCRIPTION
**Pull request by bobcao3**
... and removed an unnecessary barrier

### CODE REVIEW

File: taichi/backends/vulkan/runtime.cpp [hunk 1]
```diff
@@ -560,6 +562,19 @@ void VkRuntime::launch_kernel(KernelHandle handle, RuntimeContext *host_ctx) {
     }
   }
 
+  // If we have accumulated some work but does not require sync
+  // and if the accumulated cmdlist has been pending for some time
+  // launch the cmdlist to start processing.
+  if (current_cmdlist_) {
+    constexpr uint64_t max_pending_time = 2000;  // 3000us = 2ms
```
Comment (qiao-bo) [inline taichi/backends/vulkan/runtime.cpp:569]:
```suggestion
    constexpr uint64_t max_pending_time = 2000;  // 2000us = 2ms
```

Comment (qiao-bo) [reply taichi/backends/vulkan/runtime.cpp:569]: btw, do we need to elaborate here why 2ms?

Comment (bobcao3) [reply taichi/backends/vulkan/runtime.cpp:569]: It's just a value I just decided on. (A queue submission is on the scale of 0.1ms) Doesn't seem to cause too much overhead & pretty good at overlapping CPU work with GPU especially for the JIT scenario.

File: taichi/backends/vulkan/vulkan_device.cpp [hunk 2]
```diff
@@ -724,13 +744,14 @@ void VulkanCommandList::bind_resources(ResourceBinder *ti_binder) {
...
-    vkCmdBindDescriptorSets(buffer_->buffer, bind_point,
-                            current_pipeline_->pipeline_layout()->layout,
+    vkCmdBindDescriptorSets(buffer_->buffer, bind_point, pipeline_layout,
                             /*firstSet=*/0,
                             /*descriptorSetCount=*/1, &set->set,
                             /*dynamicOffsetCount=*/0,
                             /*pDynamicOffsets=*/nullptr);
     buffer_->refs.push_back(set);
+
+    last_bound_layout_ = pipeline_layout;
```
Comment (qiao-bo) [inline taichi/backends/vulkan/vulkan_device.cpp:754]: this seems not used anywhere?

**Review by qiao-bo — APPROVED** [review]
a few nits, LGTM

File: taichi/backends/vulkan/vulkan_device.cpp (hunk 2 above)
Comment (bobcao3) [reply taichi/backends/vulkan/vulkan_device.cpp:754]: good catch

**Review by ailzhang — APPROVED** [review]
LGTM!

**Pull request merged**